blueprints/cloud-operations/network-dashboard/cloud-function/metrics/secondarys.py

```diff
@@ -0,0 +1,273 @@
+    '''
+  cluster_secondary_dict = {}
+  node_secondary_dict = {}
+
+  # Creating cluster dict
+  # Cluster dict has subnet information
+  response_cluster = config["clients"]["asset_client"].list_assets(
+      request={
+          "parent": f"organizations/{config['organization']}",
+          "asset_types": ['container.googleapis.com/Cluster'],
+          "content_type": 'RESOURCE',
+          "page_size": config["page_size"],
+      })
+
+  for asset in response_cluster:
+    cluster_project = asset.resource.data['selfLink'].split('/')[5]
+    cluster_parent = "/".join(asset.resource.data['selfLink'].split('/')[5:10])
+    cluster_subnetwork = asset.resource.data['subnetwork']
+    cluster_service_rangeName = asset.resource.data['ipAllocationPolicy'][
+        'servicesSecondaryRangeName']
+
+    cluster_secondary_dict[f"{cluster_parent}/Service"] = {
+        "project": cluster_project,
+        "subnet": cluster_subnetwork,
+        "secondaryRange_name": cluster_service_rangeName,
+        'used_ip_addresses': 0,
+    }
+
+    for node_pool in asset.resource.data['nodePools']:
+      nodepool_name = node_pool['name']
+      node_IPrange = node_pool['networkConfig']['podRange']
+      cluster_secondary_dict[f"{cluster_parent}/{nodepool_name}"] = {
+          "project": cluster_project,
+          "subnet": cluster_subnetwork,
+          "secondaryRange_name": node_IPrange,
+          'used_ip_addresses': 0,
+      }
+
+  # Creating node dict
+  # Node dict allows 1:1 mapping of pod IP utilization, and which secondary Range it is using
+  response_node = config["clients"]["asset_client"].search_all_resources(
+      request={
+          "scope": f"organizations/{config['organization']}",
+          "asset_types": ['k8s.io/Node'],
+          "read_mask": read_mask,
+          "page_size": config["page_size"],
+      })
+
+  for asset in response_node:
+    # Node name link format:
+    # "//container.googleapis.com/projects/<PROJECT_ID>/<zones/region>/<LOCATION>/clusters/<CLUSTER_NAME>/k8s/nodes/<NODE_NAME>"
+    node_parent = "/".join(asset.name.split('/')[4:9])
+    node_name = asset.name.split('/')[-1]
+    node_full_name = f"{node_parent}/{node_name}"
+
+    for versioned in asset.versioned_resources:
+      node_secondary_dict[node_full_name] = {
+          'node_parent':
+              node_parent,
+          'this_node_pool':
+              versioned.resource['metadata']['labels']
+              ['cloud.google.com/gke-nodepool'],
+          'used_IP':
+              0
+      }
+
+  # Counting IP addresses used by pods in GKE
+  response_pods = config["clients"]["asset_client"].search_all_resources(
+      request={
+          "scope": f"organizations/{config['organization']}",
+          "asset_types": ['k8s.io/Pod'],
+          "read_mask": read_mask,
+          "page_size": config["page_size"],
+      })
+
+  for asset in response_pods:
+    pod_parent = "/".join(asset.name.split('/')[4:9])
```

Comment:
Maybe provide an example just above in a comment so we understand how the asset.name looks like and which part you extract?